### PR TITLE
[Feature] Add Engine Account support

### PR DESCRIPTION
**Pull request by jnsdls**
## PR-Codex overview
The focus of this PR is to export `EngineAccountOptions` and `engineAccount` from `wallets/engine-account.js` to `wallets.ts`.

### Detailed summary
- Export `EngineAccountOptions` and `engineAccount` from `wallets/engine-account.js` to `wallets.ts`
- Add `engineAccount` function to create an engine account with various methods for transactions, message signing, and typed data signing

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jnsdls** [comment]
* **#3584** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3584?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jnsdls and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by jnsdls** [comment]
TODO: add tests

File: packages/thirdweb/src/wallets/engine-account.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+          };
+        };
+        if (queueJSON.result.transactionHash) {
+          return {
+            transactionHash: queueJSON.result.transactionHash,
+          };
+        }
+        // arbitrary time right now
+        // wait 3s before checking again
+        await new Promise((resolve) => setTimeout(resolve, 3000));
```
Comment (joaquim-verges) [inline packages/thirdweb/src/wallets/engine-account.ts:109]: 3s is quite long no? Would do every second

**Review by joaquim-verges — APPROVED** [review]
Let's give this a spin! Cc @arcoraven @d4mr @farhanW3

**Pull request closed without merging**